### ISSUE TITLE
[FORMS] BUG: ConfirmedPasswordField inserts first password into confirm field

### ISSUE THREAD

**Issue by tomspeak** [silverstripe/silverstripe-framework#2496]
When using `ConfirmedPasswordField`, if you submit without filling in the second confirm field, it automatically adds the value of the initial password into the confirm password field when failing validation

E.G:

New Password: XYZ
Confirm New Password: [left blank]

When the user submits a form with this data, an error is thrown and they're redirected back to the page, but the Confirm New Password has been auto filled in with the New Password's value.

**Comment by dhensby** [comment]
This has regressed

**Comment by dhensby** [comment]
Test to prove bug:

```php


	public function testValueMismatch() {
		$form = new Form(
			new Controller(),
			'Form',
			new FieldList($field = new ConfirmedPasswordField('SetPassword')),
			new FieldList()
		);
		$form->loadDataFrom(array(
			'SetPassword' => array(
				"_Password" => "abc123",
				"_ConfirmPassword" => "",
			),
		));
		$this->assertNotEquals($field->children->first()->Value(), $field->children->last()->Value());
		$this->assertEquals('abc123', $field->children->first()->Value());
		$this->assertEmpty($field->children->last()->Value());

		$form->loadDataFrom(array(
			'SetPassword' => array(
				"_Password" => "abc123",
				"_ConfirmPassword" => "a1b2c3",
			),
		));
		$this->assertNotEquals($field->children->first()->Value(), $field->children->last()->Value());
		$this->assertEquals('abc123', $field->children->first()->Value());
		$this->assertEquals('a1b2c3', $field->children->last()->Value());

		$form->loadDataFrom(array(
			'SetPassword' => array(
				"_Password" => "",
				"_ConfirmPassword" => "a1b2c3",
			),
		));
		$this->assertNotEquals($field->children->first()->Value(), $field->children->last()->Value());
		$this->assertEmpty($field->children->first()->Value());
		$this->assertEquals('a1b2c3', $field->children->last()->Value());
	}
```

**Comment by tractorcow** [comment]
Needs re-confirming.

**Comment by kinglozzer** [comment]
Confirmed in 3.6.1

**Comment by tractorcow** [comment]
Yes confirmed in 4.0.0 also.

**Comment by tractorcow** [comment]
Fix for 4.1 at https://github.com/silverstripe/silverstripe-framework/pull/7594

However, the new API doesn't work for 4.0.0. What kind of compromise would you prefer for this case, @dhensby ? Either we mess around with the assigned value, or hacking in a non-public API for the DisplaysSetValue field, I would expect.

**Comment by sminnee** [comment]
OK I think that the test failure that @dhensby found, at least running it now, is a bit of an edge-case, but I've raised a PR to fix it and then we can close this off.
